src/pipeline.ipynb

```diff
@@ -0,0 +1,708 @@
+    "\n",
+    "5. [**Add Segmentation Maps**](https://github.com/ml6team/fondant/tree/main/components/segment_images): This component segments the images using the [UPerNet](https://huggingface.co/docs/transformers/model_doc/upernet) model. Each segmentation map contains segments of 150 possible categories listed [here](https://huggingface.co/openmmlab/upernet-convnext-small/blob/main/config.json#L110)."
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Environment\n",
+    "\n",
+    "#### This section checks the prerequisites of your environment. Read any errors or warnings carefully."
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "**Ensure a Python between version 3.8 and 3.10 is available**"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "import sys\n",
+    "if sys.version_info < (3, 8, 0) or sys.version_info >= (3, 11, 0):\n",
+    "    raise Exception(f\"A Python version between 3.8 and 3.10 is required. You are running {sys.version}\")"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "**Check if docker compose is installed and the docker daemon is running**"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "!docker compose version >/dev/null\n",
+    "!docker info >/dev/null"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "**Check if GPU is available**"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "import logging\n",
+    "import subprocess\n",
+    "\n",
+    "try:\n",
+    "    subprocess.check_output('nvidia-smi')\n",
+    "    logging.info(\"Found GPU, using it!\")\n",
+    "    number_of_accelerators = 1\n",
+    "    accelerator_name = \"GPU\"\n",
+    "except Exception:\n",
+    "    logging.warning(\"We recommend to run this pipeline on a GPU, but none could be found, using CPU instead\")\n",
+    "    number_of_accelerators = None\n",
+    "    accelerator_name = None"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "**Check if Fondant is installed**"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "try:\n",
+    "    import fondant\n",
+    "except ImportError:\n",
+    "    logging.warning(\"Please install Fondant from the `requirements.txt` at the root of this repository\")"
```

Comment:
I would add a cell to install the requirements.txt from the notebook. To make it really bulletproofed.